Server/ConcordiaCurriculumManager/Services/CourseGroupingService.cs

```diff
@@ -307,4 +308,37 @@ public async Task<CourseGrouping> PublishCourseGrouping(Guid commonIdentifier)
 
     private async Task<Dossier> GetDossierDetailsByIdOrThrow(Guid id) => await _dossierRepository.GetDossierByDossierId(id)
         ?? throw new NotFoundException("The dossier does not exist.");
+
+    public async Task<List<CourseGrouping?>> GetCourseGroupingsByDossierAndName(Guid dossierId, string searchQuery)
+    {
+        var dossier = await _dossierRepository.GetDossierByDossierId(dossierId);
+        if (dossier == null)
+        {
+            throw new NotFoundException($"Dossier with ID {dossierId} not found.");
+        }
+
+        var groupingsByName = await _courseGroupingRepository.GetCourseGroupingsLikeName(searchQuery);
+
+        var creationGroupingsInDossier = dossier.CourseGroupingRequests
+            .Where(req => req.RequestType == RequestType.CreationRequest && req.CourseGrouping != null && req.CourseGrouping.Name.Contains(searchQuery, StringComparison.OrdinalIgnoreCase))
+            .Select(req => req.CourseGrouping)
+            .ToList();
+
+        var combinedGroupings = groupingsByName.Concat(creationGroupingsInDossier)
+            .DistinctBy(g =>  g!.Id)
+            .ToList();
+
+        var deletionGroupingIds = dossier.CourseGroupingRequests
+            .Where(req => req.RequestType == RequestType.DeletionRequest)
+            .Select(req => req.CourseGrouping?.CommonIdentifier)
+            .Where(id => id.HasValue)
+            .Select(id => id!.Value)
+            .ToList();
+
+        var filteredGroupings = combinedGroupings
```

Comment:
The filtering still doesn't work correctly because this filtering compares it to the Id of the grouping, not its CommonIdentifier